File: private.http/Objects/HtmlPage.vb
Imports System.Text

Public Class HtmlPage

    Public Property Writer As New StringBuilder
    Public Property Source As String

    Public Sub Begin(Title As String)
        Me.WriteHeader(Title)
    End Sub

    Public Sub Close()
        WriteFooter()
        Me.Source = Writer.ToString
    End Sub

    Private Sub WriteHeader(Title As String)
        Writer.AppendLine($"<html><head><title>{Title}</title></head><body style=""font-family:verdana; font-size: 9pt;"">")
    End Sub
    Private Sub WriteFooter()
        Writer.AppendLine($"</body></html>")
    End Sub
End Class

File: private.http/Objects/HtmlPage.cs
using System.Text;

namespace @private.http
{
    public class HtmlPage
    {
        public StringBuilder Writer { get; set; } = new StringBuilder();
        public string Source { get; set; }

        public void Begin(string Title)
        {
            WriteHeader(Title);
        }

        public void Close()
        {
            WriteFooter();
            Source = Writer.ToString();
        }

        private void WriteHeader(string Title)
        {
            Writer.AppendLine($"<html><head><title>{Title}</title><link rel=\"icon\" type=\"image/x-icon\" href=\"/@res/app.ico\"></head><body style=\"font-family:verdana; font-size: 9pt;\">");
        }

        private void WriteFooter()
        {
            Writer.AppendLine($"</body></html>");
        }
    }
}